src/couch/src/couch_file.erl

```diff
@@ -264,6 +265,19 @@ rename_file(Original) ->
         Else -> Else
     end.
 
+rename_dir(RootDelDir, Original, DbName) ->
+    DbDir = binary_to_list(DbName) ++ "_design",
+    [DbPureName | _R] = lists:reverse(filename:split(binary_to_list(DbName))),
```

Comment:
The logic of above line is to get pure dbname with epochs suffix. 

Let me give example:
```
RootDelDir = /srv/view_index
DbDir = shards/e0000000-ffffffff/testdb1.1500350972_design
Original = /srv/view_index/.shards/e0000000-ffffffff/testdb1.1500350972_design
DBName = shards/e0000000-ffffffff/testdb1.1500350972
```
In this PR, we experienced two evolutions. 

First, we select recovery directory, like `filename:join([RootDelDir, ".recovery", DbDir]),`. Thus, the constructed directory looks like `/srv/view_index/.recovery/shards/e0000000-ffffffff/testdb1.1500350972_design`. This is feasible. 

However, we want to group these moved view directories and files. So you can see that current approach is `filename:join([RootDelDir, ".recovery", DbPureName, DbDir]),`. The DbPureName looks like `testdb1.1500350972` and doesn't contains `shards/e0000000-ffffffff`. Thus, for view files belonging to the same database, they can be easily managed by using such directory structure. There are two points which are considered:
  - for database which are deleted and created using same name, their epochs are different. This will not cause confusion.
  - for database with the same name from different user, in most situation, their epochs are different. Even if their epochs are the same, they can be still managed because they will be put into different subdirectory, like
  `/srv/view_index/.recovery/testdb1.1500350972/shards/e0000000-ffffffff/user1/testdb1.1500350972_design` and
  `/srv/view_index/.recovery/testdb1.1500350972/shards/e0000000-ffffffff/user2/testdb1.1500350972_design`